### ISSUE TITLE
Gmail happens the send multiple `FETCH` response for a same seq/uid

### ISSUE THREAD

**Issue by soywod** [modern-email/defects#39]
I cannot find the original context (I am pretty sure I already faced this issue in the past), but this is the most recent issue I have: https://github.com/pimalaya/himalaya/issues/498.

To give a bit more context, this happens when reading an unseen message. The first `FETCH` contains `Uid(n), Flags([Flag(Seen)]), BodyExt`, and the second contains `Uid(n), Flags([Flag(Seen)])`.

**Comment by duesee** [comment]
As IMAP is a cache-filling protocol, it seems to be a valid thing to do. The client is expected to update it's database twice in this case, I think.

Regarding `imap-client`...

Does `imap-client` emitted a warning in this case? Just curious. For `imap-client`s command-based interface I think we should merge duplicate responses. What do you think?

**Comment by soywod** [comment]
> Does `imap-client` emitted a warning in this case?

Yes it does: https://github.com/pimalaya/imap-client/commit/39d0217a58637a5f7e017e12bd8845f562bfd5b6#diff-b0a0498b0d99e8443353139a00701d7471af24e5567dda0a4e9ee7e5f23e496eR62

The difference is that before it used to replace the items from the previous response. Now it merges items, which fixes the issue. Yet is it valid to receive twice such response according to RFC?

**Comment by duesee** [comment]
I think it is. Because: If you look at IMAP as "updating your database constantly" protocol, sending data in pieces would have the same result as sending at once. But let's maybe keep this issue open in case someone else has a more definitive answer.
